### PR TITLE
MGDAPI-5877

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
# Issue link
[MGDPI-5877](https://issues.redhat.com/browse/MGDAPI-5877)

# What
Available condition type of  `MonitoringStack` is now reported in `AddonInstance`

# Verification steps
1. Checkout this branch
2.  Run 
``` 
make cluster/prepare/local
make code/run
```
3. Wait for RHOAM to install
4.  `AddonInstance` should now also report the availability of `MonitoringStack`
```
watch "oc get addoninstance addon-instance -n redhat-rhoam-operator -o json | jq '.status'
```
Expected Results
```
conditions:
    - lastTransitionTime: '2023-08-04T13:58:21Z'
      message: Heartbeat not received before timeout threshold.
      observedGeneration: 1
      reason: HeartbeatTimeout
      status: Unknown
      type: addons.managed.openshift.io/Healthy
    - lastTransitionTime: '2023-08-04T12:20:47Z'
      message: Installation complete
      observedGeneration: 1
      reason: SetupComplete
      status: 'True'
      type: addons.managed.openshift.io/Installed
    - lastTransitionTime: '2023-08-04T12:23:47Z'
      message: Core components healthy
      observedGeneration: 1
      reason: Healthy
      status: 'True'
      type: integreatly.org/Healthy
    - lastTransitionTime: '2023-08-04T12:23:47Z'
      message: All components healthy
      observedGeneration: 1
      reason: Degraded
      status: 'False'
      type: addons.managed.openshift.io/Degraded
    - lastTransitionTime: '2023-08-04T12:19:25Z'
      message: Monitoring Stack is available
      observedGeneration: 1
      reason: MonitoringStackAvailable
      status: 'True'
      type: MonitoringStackAvailable
  lastHeartbeatTime: '2023-08-04T13:57:51Z'
  observedGeneration: 1
```

### CODE REVIEW

**Comment by Patryk-Stefanski** [comment]
Commit `b0810b2618eccf97ab5c23e3868e9743db338dd3` attempts to fix failing unit tests for testing the reconciliation of the status controller in `status_controller_test.go`. The issue encountered is that the client used in the tests can't find the `MonitoringStack`.  A similar issue was encountered in  `status_controller.go` but was worked around by creating a `serverClient` which was able to find the `monitoringStack`.

Stopping work on this as don't have scope.

File: controllers/status/status_controller_test.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package status
 import (
 	"context"
 	"fmt"
+	"github.com/integr8ly/integreatly-operator/pkg/config"
```
Comment (MStokluska) [inline controllers/status/status_controller_test.go:6]: Please fix imports order

**Comment by MStokluska** [comment]
Left a couple of questions, apart from that all looks good, once resolved I will lgtm

File: controllers/status/status_controller.go [hunk 2]
```diff
@@ -51,8 +53,18 @@ func New(mgr manager.Manager, opts ...ControllerConfig) (*StatusReconciler, erro
...
+
+	client, err := k8sclient.New(restConfig, k8sclient.Options{
+		Scheme: mgr.GetScheme(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
 	return &StatusReconciler{
-		Client:              mgr.GetClient(),
```
[the remark this answers is not in the document]
Comment (carlkyrillos) [reply controllers/status/status_controller.go:55]: I forget if that change was strictly speaking necessary but I believe we changed that to bring it closer to the pattern that the [subscription_controller](https://github.com/integr8ly/integreatly-operator/blob/70d4e4a0d6d4fc9ad363837d64e06696454ae6ff/controllers/subscription/subscription_controller.go#L89), [apimanagementtenant_controller](https://github.com/integr8ly/integreatly-operator/blob/2f7b0be44e2e6ddd181808ef5a32f8949306d093/controllers/tenant/apimanagementtenant_controller.go#L43), and [user_controller](https://github.com/integr8ly/integreatly-operator/blob/3d8957c681f48430840b34c67bf562c8f36daeca/controllers/user/user_controller.go#L46) use for consistency.

**Comment by carlkyrillos** [comment]
Looks like there's an issue with prow - `build error: creating build container: reading signatures: downloading signatures for sha256 in registry.ci.openshift.org/openshift/release: received unexpected HTTP status: 504 Gateway Time-out`

I'll try running the tests again later today

**Pull request merged**